subsystems/TypeHandlerLibrary/src/test/java/org/terasology/persistence/typeHandling/FutureTypeHandlerTest.java

```diff
@@ -77,4 +53,29 @@ public void testRecursiveType() {
 
         assertEquals(typeHandler, future.typeHandler);
     }
+
+    private static final class RecursiveType<T> {
+        final T data;
+        final List<RecursiveType<T>> children;
+
+        @SafeVarargs
+        private RecursiveType(T data, RecursiveType<T>... children) {
+            this.data = data;
+            this.children = Lists.newArrayList(children);
+        }
+    }
+
+    @edu.umd.cs.findbugs.annotations.SuppressFBWarnings
```

Comment:
why is this necessary?

Reply:
added a comment. message that one should consider making the class static. but the test has it non-static.

Reply:
> but the test has it non-static.

Sorry, you'll have to elaborate that a bit more, I don't follow.

Reply:
updated the comment @jdrueckert .

Reply:
Based on the reasoning in my comment above (line 68), I'd propose to adjust this as follows and remove the comment you added (lines 68-70):
```suggestion
    @edu.umd.cs.findbugs.annotations.SuppressFBWarnings(value = "SIC_INNER_SHOULD_BE_STATIC", justification = "Test code is not performance-relevant, flagged inner ResultCaptor class is a mock with dynamic behavior and cannot be static.")
```

`value` filters the exact spotbugs finding that should be suppressed, which avoids suppressing other (future) findings that might be valid
`justification` documents why we can ignore this finding here in a concise way and as part of the suppression annotation